system/include/emscripten/promise.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#pragma once
+
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// An opaque handle to a JS Promise object.
+typedef struct em_promise_t {
+  void* handle;
+} em_promise_t;
+
+typedef enum em_promise_result_t {
+  EM_PROMISE_FULFILL,
```

Comment:
Why "fulfill" and not "resolve" (the standard JS term)?

Reply:
Fulfill and resolve mean different things ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#:~:text=You%20will%20also,well.%20For%20example%3A)) and I precisely mean "fulfill" here.